docs/notebook_validation.py

```diff
@@ -0,0 +1,104 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import glob
+import os
+import re
+import sys
+from pathlib import Path
+import subprocess
+from shutil import which
+
+if which('jupyter') is None:
+    print("Please install jupyter, e.g. with `pip install notebook`.")
+    exit(1)
+
+
+def read_available_backends():
+    available_backends = sys.stdin.readlines()
+    return [backend.strip() for backend in available_backends]
+
+
+def validate(notebook_filename, available_backends):
+    with open(notebook_filename) as f:
+        lines = f.readlines()
+    for notebook_content in lines:
+        match = re.search('set_target[\\\s\(]+"(.+)\\\\"[)]', notebook_content)
+        if match and (match.group(1) not in available_backends):
+            return False
+    return True
+
+
+def execute(notebook_filename):
+    notebook_filename_out = notebook_filename.replace('.ipynb',
+                                                      '.nbconvert.ipynb')
+    try:
+        subprocess.run([
+            "jupyter", "nbconvert", "--to", "notebook", "--execute",
+            notebook_filename
+        ],
+                       check=True)
+        os.remove(notebook_filename_out)
+        return True
+    except subprocess.CalledProcessError:
+        print('Error executing the notebook "%s".\n\n' % notebook_filename)
+        return False
+
+
+def print_results(success, failed, skipped=[]):
+    if success:
+        print("Success! The following notebook(s) executed successfully:\n" +
+              " ".join(success))
+
+    if failed:
+        print("Failed! The following notebook(s) raised errors:\n" +
+              " ".join(failed))
+
+    if skipped:
+        print("Skipped! The following notebook(s) skipped:\n" +
+              " ".join(skipped))
+
+    if not failed and not skipped:
+        print("Success! All the notebook(s) executed successfully.")
```

Comment:
A non-zero exit code will make the workflow fail. Correspondingly, you want to call exit(1) when there was a notebook that failed.
```suggestion
        print("Success! All the notebook(s) executed successfully.")
    elif failed:
```